src/main.c

```diff
@@ -660,37 +658,11 @@ void get_home_path(app_widgets *widgets) {
     gtk_text_buffer_set_text(GTK_TEXT_BUFFER(g_text_revealer), "User error", -1);
     gtk_revealer_set_reveal_child (GTK_REVEALER(widgets->g_revealer), TRUE);
   }
-
-  // if upp file is placed in ~/.local/bin and ~/.local/bin doesn't exist in path (old Ubuntu/Debian bug), add it to path
-  snprintf(localpath, sizeof(localpath), "%s/.local/bin", homedir);
-  snprintf(localupppath, sizeof(localupppath), "%s/upp", localpath);
-  if (access(localupppath, F_OK) != -1) {    
-    char *currenv = strdup(getenv("PATH"));
-    if(strstr(currenv, localpath) == NULL) {
-      char newenv[2048];
-      snprintf(newenv, sizeof(newenv),"%s:%s", currenv, localpath);
-      setenv("PATH", newenv, 1);
-    }
-    free(currenv);
-  }
 }
 
-void get_upp_path(app_widgets *widgets) {
-  FILE *fupppath = popen("which upp", "r");
-    if (fupppath) {
-      if (fgets(upppath, sizeof(upppath), fupppath)){
-        size_t len = strlen(upppath);
-        if (len > 0 && upppath[len-1] == '\n') {
-          upppath[--len] = '\0';
-        }
-        printf("UPP path is %s\n", upppath);
-      }
-      else {
-        gtk_text_buffer_set_text(GTK_TEXT_BUFFER(g_text_revealer), "UPP module not found, install with pip", -1);
-        gtk_revealer_set_reveal_child (GTK_REVEALER(widgets->g_revealer), TRUE);
-      }
-    }
-    pclose(fupppath);
+void set_upp_path(app_widgets *widgets) {
+  snprintf(upppath, sizeof(upppath), "%s/.local/bin/venv/bin/upp", getenv("HOME"));
```

Comment:
This breaks my setup by not finding the executable.

Reply:
ok I used `git clone https://github.com/azeam/powerupp.git` & just ran powerupp.desktop directly from the folder it grabbed & this is working for me again now with my 6800XT. Thanks! :)